### ISSUE TITLE
Loss oscillation

### ISSUE THREAD

**Issue by FTKyaoyuan** [lucidrains/imagen-pytorch#246]
hi
My loss looks strange. Is it normal？
![image](https://user-images.githubusercontent.com/59943491/196833093-10a0e543-fc71-4569-b13f-d2c524733745.png)

**Comment by Feanor007** [comment]
I got similar loss curves. Did you observe any visual improvement in your synthesized image?

**Comment by FTKyaoyuan** [comment]
> 我得到了类似的损失曲线。您是否在合成图像中观察到任何视觉改善？

I'm training my first unet so I'm also sampling for

**Comment by FTKyaoyuan** [comment]
> I got a similar loss curve. Did you observe any visual improvement in your synthesized image?

I'm training my first unet so I'm also sampling for 
I will start training my second unet tomorrow

**Comment by FTKyaoyuan** [comment]
> I got a similar loss curve. Did you observe any visual improvement in your synthesized image?

What did your training result look like?

**Comment by Feanor007** [comment]
I trained my model to synthesize gray scale images (channels = 1). 
This is what it looks like after 5000 steps:
![Screen Shot 2022-10-20 at 10 21 58](https://user-images.githubusercontent.com/25071118/196909884-ceacf439-92ed-4194-93a9-bddd142a381d.jpg)
This is what it looks like now (12000 steps):
![Screen Shot 2022-10-20 at 10 23 12](https://user-images.githubusercontent.com/25071118/196910166-97e26ce5-07fe-4b46-945d-02bc5de226e1.jpg)

You can see there are improvements, but still not perfect for a thoracic CT scan.

**Comment by Feanor007** [comment]
Also, the training was really slow to me. It took more than 24 hours to reach 10000 steps

**Comment by FTKyaoyuan** [comment]
> Also, the training was really slow to me. It took more than 24 hours to reach 10000 steps. Is this same for your?

![image](https://user-images.githubusercontent.com/59943491/196911420-5ad40c72-bc0e-4505-a1f2-9d0242c3d625.png)
I trained about 450k setp for 24 hours

**Comment by Feanor007** [comment]
what does your result look like?

**Comment by FTKyaoyuan** [comment]
I don't know yet because I only trained unet1
I have two Unet
I want to view the results after training two unets

**Comment by FTKyaoyuan** [comment]
tomorrow i will train unet2

**Comment by FTKyaoyuan** [comment]
The GPU I am using is 3080
And yours？

**Comment by Feanor007** [comment]
> The GPU I am using is 3080
> And yours？

Nvidia Titan X. Looking forward to your results!

**Comment by Feanor007** [comment]
> > > The GPU I am using is 3080
> > > And yours？
> > 
> > 
> > Nvidia Titan X. Looking forward to your results!
> 
> you Unet number is ??

Just one U-Net, the image size is 64 * 64 * 32.  Will do the superresolution later

**Comment by FTKyaoyuan** [comment]
ok 
If there is a result, I will let you know as soon as possible
Your results look like they should be fine if you keep training. Try training for another 24 hours
You can try changing lr to 1e-5

**Comment by Feanor007** [comment]
> ok
> If there is a result, I will let you know as soon as possible
> Your results look like they should be fine if you keep training. Try training for another 24 hours
> You can try changing lr to 1e-5

Thanks. If you don't mind, can you take look at issue #245 ? Perhaps there is something wrong with the setup that makes my training so slow?

**Comment by FTKyaoyuan** [comment]
![image](https://user-images.githubusercontent.com/59943491/197120459-706531ef-de80-449f-b92c-4633abf39392.png)
hi


> > 好的，如果有结果，我会尽快通知您，如果您继续训练，您的结果看起来应该没问题。尝试再训练 24 小时 您可以尝试将 lr 更改为 1e-5
> 
> 谢谢。如果你不介意，你能看看问题[#245吗](https://github.com/lucidrains/imagen-pytorch/issues/245)？也许设置有问题，使我的训练如此缓慢？

This is my image. It looks like it has an image, but I think it has nothing to do with my text, and the resolution is somewhat low. I can't even see the content clearly

**Comment by FTKyaoyuan** [comment]
![image](https://user-images.githubusercontent.com/59943491/197120678-4f28d9e4-e138-46af-ab54-e93ba4df6aea.png)
![image](https://user-images.githubusercontent.com/59943491/197120707-f7ec5364-169d-447b-bceb-e498c5170a74.png)

**Comment by FTKyaoyuan** [comment]
![image](https://user-images.githubusercontent.com/59943491/197120724-78ab808b-5901-4cd1-8818-d3fdeef10b1a.png)

**Comment by Feanor007** [comment]
Maybe train the U-Nets separately? Easier for debugging

**Comment by Feanor007** [comment]
> ![image](https://user-images.githubusercontent.com/59943491/197120459-706531ef-de80-449f-b92c-4633abf39392.png) hi
> 
> > > 好的，如果有结果，我会尽快通知您，如果您继续训练，您的结果看起来应该没问题。尝试再训练 24 小时 您可以尝试将 lr 更改为 1e-5
> > 
> > 
> > 谢谢。如果你不介意，你能看看问题[#245吗](https://github.com/lucidrains/imagen-pytorch/issues/245)？也许设置有问题，使我的训练如此缓慢？
> 
> This is my image. It looks like it has an image, but I think it has nothing to do with my text, and the resolution is somewhat low. I can't even see the content clearly

By the way, how did you sample using two U-Nets without getting OOM error?

**Comment by FTKyaoyuan** [comment]
> 也许单独训练U-Nets？更易于调试

I still don't quite understand the purpose of the two unets to achieve different resolutions. For example, 64*64 uses the first one, and 128*128 uses the second one?

> Maybe train the U-Nets separately? Easier for debugging

I still don't quite understand the purpose of the two unets to achieve different resolutions. For example, 64*64 uses the first one, and 128*128 uses the second one?

**Comment by FTKyaoyuan** [comment]
> > ![image](https://user-images.githubusercontent.com/59943491/197120459-706531ef-de80-449f-b92c-4633abf39392.png) hi
> > > > 好的，如果有结果，我会尽快通知您，如果您继续训练，您的结果看起来应该没问题。尝试再训练 24 小时 您可以尝试将 lr 更改为 1e-5
> > > 
> > > 
> > > 谢谢。如果你不介意，你能看看问题[#245吗](https://github.com/lucidrains/imagen-pytorch/issues/245)？也许设置有问题，使我的训练如此缓慢？
> > 
> > 
> > This is my image. It looks like it has an image, but I think it has nothing to do with my text, and the resolution is somewhat low. I can't even see the content clearly
> 
> By the way, how did you sample using two U-Nets without getting OOM error?

I didn't specify unet when I sampled, no error was reported

**Comment by BBaDBoYY** [comment]
And how many epochs have you trained, which dataset are you using

**Comment by FTKyaoyuan** [comment]
> Hello，Do you have a model trained now? I would like to ask you about the specific value of the Unet parameter. The results I get so far are not good.The images I've gotten so far are rather blurry.

I used two unet datasets for mscoco but I feel that my resolution is relatively small, so the generated effect does not look good

**Comment by FTKyaoyuan** [comment]
> And how many epochs have you trained, which dataset are you using

5  epoch

**Comment by FTKyaoyuan** [comment]
> And how many epochs have you trained, which dataset are you using

can i see your results？

**Comment by BBaDBoYY** [comment]
![1](https://user-images.githubusercontent.com/115871584/198274915-9a1c1e9b-aa1f-4add-85bc-d92a04c86021.png)
text:a big building with a clock at the top of it 
![1666856724998](https://user-images.githubusercontent.com/115871584/198275165-d2e15ed6-e728-4bc6-aa31-73d2e154bc76.png)
@FTKyaoyuan

**Comment by BBaDBoYY** [comment]
I trained the model on the COCO2017 dataset with a total of 150,000 data pairs.

**Comment by FTKyaoyuan** [comment]
> ![1](https://user-images.githubusercontent.com/115871584/198274915-9a1c1e9b-aa1f-4add-85bc-d92a04c86021.png) text:a big building with a clock at the top of it ![1666856724998](https://user-images.githubusercontent.com/115871584/198275165-d2e15ed6-e728-4bc6-aa31-73d2e154bc76.png) @FTKyaoyuan

Maybe you should fill and then resize otherwise the graphics will be distorted

**Comment by BBaDBoYY** [comment]
> > ![1](https://user-images.githubusercontent.com/115871584/198274915-9a1c1e9b-aa1f-4add-85bc-d92a04c86021.png) text:a big building with a clock at the top of it ![1666856724998](https://user-images.githubusercontent.com/115871584/198275165-d2e15ed6-e728-4bc6-aa31-73d2e154bc76.png) @FTKyaoyuan
> 
> Maybe you should fill and then resize otherwise the graphics will be distorted

thank you for your reply

**Comment by zhangnan-hust** [comment]
How is your dataset loaded?

**Comment by statyui** [comment]
> > > ![1](https://user-images.githubusercontent.com/115871584/198274915-9a1c1e9b-aa1f-4add-85bc-d92a04c86021.png) text:a big building with a clock at the top of it ![1666856724998](https://user-images.githubusercontent.com/115871584/198275165-d2e15ed6-e728-4bc6-aa31-73d2e154bc76.png) @FTKyaoyuan
> > 
> > 
> > Maybe you should fill and then resize otherwise the graphics will be distorted
> 
> thank you for your reply

My generated image is similar to yours, did you solve it?

**Comment by BBaDBoYY** [comment]
> 
No, I finally gave up on that idea
